S3/ConnMan.py

```diff
@@ -51,9 +51,34 @@ def _ssl_context():
         return context
 
     @staticmethod
+    def match_hostname_aws(cert, hostname, e):
+        san = cert.get('subjectAltName', ())
+        for key, value in san:
+            if key == 'DNS':
+                if value.startswith('*.s3') and value.endswith('.amazonaws.com') and hostname.endswith('.amazonaws.com'):
```

Comment:
This is correct, but is going to look mighty suspicious to some security person someday.  A comment linking back to the original issue might help.